CMakeLists.txt

```
@@ -948,7 +948,24 @@ if(CONFIG_USERSPACE)
 endif()
 
 get_property(CSTD GLOBAL PROPERTY CSTD)
-set_ifndef(CSTD c99)
+if(NOT DEFINED CSTD)
+  if(CONFIG_STD_C23)
+    set(CSTD c2x)
+  elseif(CONFIG_STD_C17)
+    set(CSTD c17)
+  elseif(CONFIG_STD_C11)
+    set(CSTD c11)
+  elseif(CONFIG_STD_C99)
+    set(CSTD c99)
+  elseif(CONFIG_STD_C90)
+    set(CSTD c90)
+  else()
+    assert(0 "Unreachable code. Expected C standard to have been chosen.")
```

nit: `assert` is a Zephyr-specific macro, which isn't needed here.
```suggestion
    message(FATAL_ERROR "Unreachable code. Expected C standard to have been chosen.")
```